packages/ledger/src/lib/ledger.spec.ts

```diff
@@ -75,56 +97,107 @@ afterEach(() => {
+    expect(ledgerClient.connect).toHaveBeenCalledTimes(1);
+    expect(ledgerClient.getPublicKey).toHaveBeenCalledTimes(1);
+    expect(storage.setItem).toHaveBeenCalledWith("accounts", [
       {
         accountId,
         derivationPath,
         publicKey,
       },
     ]);
-    expect(ledgerClient.connect).toHaveBeenCalled();
-    expect(ledgerClient.setScrambleKey).toHaveBeenCalled();
-    expect(ledgerClient.on).toHaveBeenCalled();
   });
 });
 
 describe("getAccounts", () => {
-  // TODO: Need to mock fetching for account id.
-  it.skip("returns account objects", async () => {
-    const accountId = "accountId";
+  it("returns account objects", async () => {
+    const accountId = "amirsaran.testnet";
     const { wallet } = await createLedgerWallet();
     await wallet.signIn({
-      contractId: "test.testnet",
-      derivationPaths: ["derivationPath"],
+      derivationPaths: ["44'/397'/0'/0'/1'"],
+      contractId: "guest-book.testnet",
     });
     const result = await wallet.getAccounts();
     expect(result).toEqual([{ accountId }]);
   });
+  it("returns empty list because not connected", async () => {
+    const { wallet } = await createLedgerWallet();
+    const result = await wallet.getAccounts();
+    expect(result).toEqual([]);
+  });
 });
 
-// describe("signAndSendTransaction", () => {
-//   it("signs and sends transaction", async () => {
-//     const { wallet, authData } = await createLedgerWallet();
-//     await wallet.signIn({
-//       accountId: authData.accountId,
-//       derivationPaths: [authData.derivationPath],
-//     });
-//     const result = await wallet.signAndSendTransaction({
-//       signerId: "amirsaran.testnet",
-//       receiverId: "guest-book.testnet",
-//       actions: [],
-//     });
-//     expect(result).toEqual(null);
-//   });
-// });
+describe("signAndSendTransaction", () => {
+  it("signs and sends transaction", async () => {
+    const { wallet, ledgerClient } = await createLedgerWallet();
+    await wallet.signIn({
+      derivationPaths: ["44'/397'/0'/0'/1'"],
+      contractId: "guest-book.testnet",
+    });
+    await wallet.signAndSendTransaction({
+      signerId: "amirsaran.testnet",
+      receiverId: "guest-book.testnet",
+      actions: [],
+    });
+    expect(ledgerClient.sign).toHaveBeenCalled();
+  });
+});
+
+describe("signAndSendTransactions", () => {
+  it("signs and sends only one transaction", async () => {
+    const { wallet, ledgerClient } = await createLedgerWallet();
+    await wallet.signIn({
+      derivationPaths: ["44'/397'/0'/0'/1'"],
+      contractId: "guest-book.testnet",
+    });
+    const transactions: Array<Transaction> = [
+      {
+        signerId: "amirsaran.testnet",
+        receiverId: "guest-book.testnet",
+        actions: [],
+      },
+    ];
+    const result = await wallet.signAndSendTransactions({
+      transactions: transactions,
+    });
+    expect(ledgerClient.sign).toHaveBeenCalled();
+    expect(result.length).toEqual(1);
```

Comment:
We should reference `transactions.length` to show the link and avoid issues with these tests if the input data was changed to maybe 2 transactions